src/connection.js

```diff
@@ -522,7 +522,10 @@ class Connection extends EventEmitter {
     this.socket.on('close', this.socketClose);
     this.socket.on('end', this.socketEnd);
     this.messageIo = new MessageIO(this.socket, this.config.options.packetSize, this.debug);
-    this.messageIo.on('data', (data) => { this.dispatchEvent('data', data); });
+    this.messageIo.on('data', (data) => {
+      this.clearRequestTimer();
```

Comment:
This technically belongs under SENT_CLIENT_REQUEST.data. Can't come up with a case where this breaks anything though...